**Привет, Илья!** 👋

Меня зовут Артем Пиляй. Я буду проверять твой проект. В рамках корпоративной этики мы общаемся на «ты». Но если лучше на «Вы», то нужно обозначить это сразу.
 
В тетрадке есть комментарии к проекту, они будут в отдельных ячейках Markdown.
 
<div class="alert alert-danger">

**Артём:** ✋ Такой комментарий **нужно исправить** обязательно, он критически влияет на удачное выполнение проекта.

</div>

<div class="alert alert-warning">

**Артём:** 👉 Если таких комментариев в проекте **не больше трех**, то проект будет принят. Но есть два исключения, при которых я приму проект с любым количеством желтых комментариев, красные нужно исправить все:
    
* если у тебя дедлайн, тогда оставь в начале 👆 проекта комментарий «Дедлайн, специальность (DA/DS/DA+/DS+), и номер когорты»;
* если ты считаешь, что для успешного трудоустройства и для успешной работы в команде тебе это не нужно, тогда оставь в начале проекта комментарий «Не нужно».
</div>

<div class="alert alert-success">

**Артём:** 👍 А это **самый главный комментарий**, значит все сделано хорошо. 😉

</div>

<div class="alert alert-info">

**Артём:** ✍️ В таких комментариях я оставляю подсказки как сделать **твой проект ещё лучше**.

</div>

---

**Распространенные ошибки**

<div class="alert alert-danger">

**Артём:** ✋ Не перемещай и не удаляй мои комментарии. Лучше, чтобы твои ответы на мои комментарии шли по очереди, как сообщения в мессенджере. Это ускорит проверку.
    
</div>

<div class="alert alert-danger">

**Артём:** ✋ Проверь проект на работоспособность перед отправкой на проверку. Это важный навык, который тебе пригодится как на собеседовании, при выполнении тестового задания, так и перед совещанием у руководства компании. Помни, ревьеры не вносят изменения в код, даже если это путь к датасету или ячейка в неверном формате (code/mardown).  
    
</div>

<div class="alert alert-info">

**Артём:** ✍️ Кстати, смотри как можно сделать, чтобы датасет грузился и локально, и в JupyterHub:

```py
try:
    df = pd.read_csv('C:/Users/Student/Desktop/dataset.csv')
except:
    df = pd.read_csv('/datasets/dataset.csv')
```
    
Используя эту конструкцию в будущих проектах, ты будешь спокойно отправлять проект на проверку зная, что он будет работать.
    
</div>

<div class="alert alert-danger">

**Артём:** ✋ Проект должен быть структурирован. Где названия разделов и подразделов будут записаны в ячейках Markdown.
    
</div>

<div class="alert alert-danger">

**Артём:** ✋ Если с твоим проектом будет знакомиться другой человек, например рекрутер или старший дата специалист, в нем должны быть описаны поставленные задачи, содержание датасета, промежуточные и финальные выводы. Как правило, заказчик читает только финальный вывод, поэтому важно добавить информацию о проведенном исследовании, применить форматирование текста, в общем, сделать красиво, чтобы было понятно, что проведено настоящее исследование и выводу можно верить. 

</div>  

<div class="alert alert-warning">
    
**Артём:** 👉 Длинный код и комментарии нужно переносить на новую строку. Согласись, коллеги не оценят, если им придется тратить время на пролистывание ячеек. Согласно международному стандарту [PEP8](https://www.python.org/dev/peps/pep-0008/) длина строки не должна превышать 79 символов.
Вот как можно перенести код на новую строку:

```py
# Переносить можно код внутри скобки
example = (2 *
           2)
```
Конечно, я не считаю сколько символов в строке, главное, чтобы код или комментарий не ушел за пределы ячейки (зависит от разрешения экрана). Если ушел, то это желтый комментарий для каждой нечитаемой ячейки.
    
</div>

<div class="alert alert-warning">
    
**Артём:** 👉 Тимлид, старший дата специалист или ревьюер ценят твое и своё время, подсвечивай свои комментарии в ячейках Markdown. Шаблон я тебе оставил ниже. Подробнее про синтаксис Markdown, с примерами, можно почитать [здесь](https://paulradzkov.com/2014/markdown_cheatsheet/?).
    
</div>

<div class="alert alert-success">

**Артём:** 👍 Буду рад, если воспользуешься моими советами. Они тебе помогут при трудоустройстве и при работе в команде. Если в ходе проведения исследования возникнут вопросы, не стесняйся задавай. 🙂
</div>

<div class="alert alert-info">

**Артём:** 🏁 [Итоги ревью здесь](#results)
</div>


**Поехали!** 🚀 👨‍🚀
<a id="start"></a>

<div style="border:solid green 2px; padding: 20px">

**Имя:** Это шаблон для тебя. Такой комментарий можно быстро найти.

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)
print(data.head())

   children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                   

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
no_refund = data.groupby('children').agg({'debt':['count','sum']}) #группируем таблицу и считаем сколько
result_no_refund = no_refund['debt']['sum'] / no_refund['debt']['count']*100 #Считаем процент 

result = pd.concat([no_refund, result_no_refund], axis=1).reindex(no_refund.index) 
result.columns = ['Всего кредитополучателей', 'Всего должников', 'Доля задолженности']
result = result.drop(columns =['Всего должников']) 

result




Всего кредитополучателей  Доля задолженности
children                                              
0                            14091            7.543822
1                             4808            9.234609
2                             2052            9.454191
3                              330            8.181818
4                               41            9.756098
5                                9            0.000000

<a id="cell_1"></a>
<div class="alert alert-danger">

**Артём:** ✋ Сделай сводную таблицу в которой буду следующая информация:
        
1. Название категории;
2. Количество клиентов в категории;
3. Доля задолженности.
    
Это важно сделать в шагах 3.1, 3.2, 3.3 и 3.4. Подробнее можно почитать [здесь](https://dfedorov.spb.ru/pandas/Подробное%20руководство%20по%20группировке%20и%20агрегированию%20с%20помощью%20pandas.html?ysclid=lcmm6rtg2w602526506)    
    
Высшем пилотажем будет если проценты сразу выведешь в правильном формате, то есть умножишь на сто.
    
</div>

<div class="alert alert-success">

**Артём:** 👍👍 Отлично.

</div>

**Вывод:** Самый хорошие заемщики это люди без детей (это хорошо так-как это самая большая группа) или с 5-ю детьми! Последних настолько мало что можно пренебречь.
Процент не возврата кредита у людей с 1,2,4 детьми практически одинаков и равен в среднем 9.5, в нашей выборке можно считать что это группа риска! у семей с 3 детьми процент невозврата чуть ниже 8.2, но таких семей не так уж и много!

<div class="alert alert-success">

**Артём:** 👍 Правильно.

</div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:
no_refund=data.groupby('family_status').agg({'debt':['count','sum']})

result_no_refund=no_refund['debt']['sum']/no_refund['debt']['count']*100

result = pd.concat([no_refund, result_no_refund], axis=1).reindex(no_refund.index) 
result.columns = ['Всего кредитополучателей', 'Всего должников', 'Доля задолженности']
result = result.drop(columns =['Всего должников']) 

result

Всего кредитополучателей  Доля задолженности
family_status                                                      
Не женат / не замужем                      2796            9.763948
в разводе                                  1189            7.064760
вдовец / вдова                              951            6.624606
гражданский брак                           4134            9.313014
женат / замужем                           12261            7.560558

**Вывод:** Самая большая категория заёмщиков это женатые люди 12261 человек, они же практически самые ответственные платильщики всего 7.5% просрочки! люди живущие в гражданском браке вторая группа кредитующихся 4134 и при этом почти с самым высоким процентом не возврата 9,3%! Вдовцы самые ответственные 6,6% просрочки но их всего 951 заемщик.


<div class="alert alert-success">

**Артём:** 👍 Молодец.

</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [24]:
no_refund = data.groupby('total_income_category').agg({'debt':['count','sum']})
result_no_refund = no_refund['debt']['sum']/no_refund['debt']['count']*100

result = pd.concat([no_refund, result_no_refund], axis=1).reindex(no_refund.index) 
result.columns = ['Всего кредитополучателей', 'Всего должников', 'Доля задолженности']
result = result.drop(columns =['Всего должников']) 

result

Всего кредитополучателей  Доля задолженности
total_income_category                                              
A                                            25            8.000000
B                                          5014            7.060231
C                                         15921            8.498210
D                                           349            6.017192
E                                            22            9.090909

**Вывод:** 
Самые благонадёжные платильщики 6% категории D с доходом до 50 т.р. их всего 349 человек. Самые не благонадежные это заемщики с минимальным доходом до 30 т. р. 9% просрочки. Люди с доходом до 200 т.р. имеют процент невозврата 8.4% это плохо так как это основная масса заёмщиков!


<div class="alert alert-success">

**Артём:** 👍 Если закрыть глаза на то, что выборка не репрезентативна, то все корректно.
    
</div>

<div class="alert alert-info">

<b>Артём: ✍️</b> Дам тебе совет на будущее, как лучше проводить категоризацию количественных данных.
    
Сейчас самая маленькая категория в 728 раз меньше самой большой. Это может отразиться на результатах исследования. Когда мы делим на категории количественные данные, то лучше их делать примерно одинаковыми. Для этого можно воспользоваться методом `qcut()`. Пример:
    
```py
pd.qcut(df['total_income'], 4)
```
    
```
Name: total_income, Length: 21453, dtype: category
Categories (4, interval[float64]): [(20666.999, 107620.0]  (107620.0, 142594.0]  (142594.0, 195818.0]  (195818.0, 2265604.0]]
```
                                                                                                               
Что внутри: 
    
```py
pd.qcut(датафрейм['столбец'], количество_категорий)
``` 
    
Итого у нас 4 категории:
<ul>
  <li>меньше 107_620</li>
  <li>от 107_620 до 142_594</li>
  <li>от 142_594 до 195_818</li>
  <li>от 195_818</li>
</ul>
    
---
Дополнительно:<br>
<a href="https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.qcut.html"
   target="_blank">
  📖 Официальная документация.
</a><br>
<a href="https://dfedorov.spb.ru/pandas/Разделение%20данных%20в%20Pandas%20с%20помощью%20qcut%20и%20cut.html?"
   target="_blank">
  📖 Статья на русском.
</a>
</div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [25]:
no_refund = data.groupby('purpose_category').agg({'debt':['count','sum']})
result_no_refund = no_refund['debt']['sum']/no_refund['debt']['count']*100

result = pd.concat([no_refund, result_no_refund], axis=1).reindex(no_refund.index) 
result.columns = ['Всего кредитополучателей', 'Всего должников', 'Доля должников']
result = result.drop(columns =['Всего должников']) 

result

Всего кредитополучателей  Доля должников
purpose_category                                                  
операции с автомобилем                        4279        9.347978
операции с недвижимостью                     10751        7.255139
получение образования                         3988        9.252758
проведение свадьбы                            2313        7.911803

**Вывод:** 
Операции с недвижимостью самая большая причина заема средств 10751 человек с самым маленьким процентом не отдачи заема средств!
На проведение свадьбы самое не популярная категория заема 2313 человек, и всего 8% не возврата займа. "Операции с автомобилем" и "получение образования" имеют премерно одинаковое количество не возвратов!

<div class="alert alert-success">

**Артём:** 👍 Хорошо.
    
</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 
days_employed       2174
total_income        2174
Значения пропусков одинаковые! "Количество отработанных дней" и "общий доход" возможно это люди которые работают "в черную" или по иным причинам не хотят разглашать свой доход.

<div class="alert alert-warning">
    
**Артём:** 👉 Хорошо, а какие еще варианты?
    
</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 
Медианное значение лучше отражает выборку для количественных переменных так как не учитывает разброс данных.

<div class="alert alert-success">

**Артём:** 👍 Верно.
    
</div>

### Шаг 4: общий вывод.

Напишите ваш общий вывод.

В целом процентный разброс не возврата кредитов не так велик, максимальный это 9.7% минимальный 6%. Думаю самыми надежными заемщиками будут женатые люди без детей которые берут кредит на операции с недвижимостью и скорее всего это самая большая группа людей.

Новый вывод.

    В целом процентный разброс не возврата кредитов не так велик, максимальный это 9.7% минимальный 6%. Думаю самыми надежными заемщиками будут женатые люди без детей которые берут кредит на операции с недвижимостью, это самая большая группа людей.

    На основе предоставленных заказчиком данных мы провели исследование и определели зависимости  по следующим категориям заёмщиков наличее детей, семейное положение, уровень дохода и цели кридита.

	                       Всего кредитополучателей	Доля должников
purpose_category		
операции с автомобилем	          4279	             9.3
операции с недвижимостью	      10751	             7.2
получение образования	          3988	             9.2
проведение свадьбы	              2313	             7.9

    Из преведенных данных мы видим что самые ответственные заемщики это люди без детей 7.5% , это самая большая группа из исследуемых групп  14091 человек! Процент не возврата кредита у людей с 1,2,4 детьми практически одинаков и равен в среднем 9.5%, в нашей выборке можно считать что это группа риска таких заемщиков более 7000 человек! у семей с 3 детьми процент невозврата чуть ниже 8.2%, но таких заёмщиков всего 330 человек!
    
    	                   Всего кредитополучателей	Доля задолженности
family_status		
Не женат / не замужем	   2796	                    9.8
в разводе	               1189	                    7.1
вдовец / вдова	           951	                    6.6
гражданский брак	       4134	                    9.3
женат / замужем	           12261	                7.6

    Из преведенных данных мы видим что самая большая категория заёмщиков это женатые люди 12261 человек, они же практически самые ответственные платильщики всего 7.5% просрочки! люди живущие в гражданском браке вторая группа кредитующихся 4134 и при этом почти с самым высоким процентом не возврата 9,3%! Вдовцы самые ответственные 6,6% просрочки но их всего 951 заемщик.

	                       Всего кредитополучателей	Доля задолженности
total_income_category		
A	                       25	                    8.0
B	                       5014	                    7.0
C	                       15921	                8.5
D	                       349	                    6.0
E	                       22	                    9.1

    Из преведенных данных мы видим что самые благонадёжные платильщики 6% категории D с доходом до 50 т.р. их всего 349 человек. Самые не благонадежные это заемщики с минимальным доходом до 30 т. р. 9%. Люди с доходом до 200 т.р. имеют процент невозврата 8.4% стоит обратить на это внимание так как это основная масса заёмщиков!
    
                           Всего кредитополучателей	Доля должников
purpose_category		
операции с автомобилем	   4279	                    9.3
операции с недвижимостью   10751	                7.2
получение образования	   3988	                    9.2
проведение свадьбы	       2313	                    7.9

    Операции с недвижимостью самая большая причина заема средств 10751 человек, и это самые ответственные платильщик всего 7.2%! Проведение свадьбы самое не популярная категория заема 2313 человек, и всего 8% не возврата займа. "Операции с автомобилем" - 4279 человек и "получение образования" - 3988 человек, имеют премерно одинаковое количество не возвратов 9,3% и 9,2% соответственно!


<a id="cell_1000"></a>
<div class="alert alert-danger">

**Артём:** ✋ Проект пишется для заказчика, он будет читать финальный вывод и только его. Добавь информации о полученных результатах, разбей по пунктам, сделай красиво, чтобы было понятно, что проведено настоящее исследование и выводу можно верить. Учитывай для кого проведено исследование, в банковской сфере даже сотые доли процента играют роль, но в нашем можно ограничиться десятыми. Помни, вывод – это дата специалиста. 😉


</div>

<div class="alert alert-success">

**Артём:** 👍👍 Шикарный вывод.

</div>

## 🏁 Итоги ревью
<a id="results"></a>
[В начало](#start)

<div class="alert alert-danger">

**Артём:** ✋ Илья, у тебя хорошая работа, с удовольствием ее изучал. Осталось дополнить финальный вывод и построить сводные таблицы и можно идти к заказчику. Молодец!
    
Жду новую версию проекта. 🙂

</div>

<div class="alert alert-success">

**Артём:** ✌️ Илья, спасибо за работу и внесенные корректировки! Критических замечаний нет.

Поздравляю, **проект принят!** 🥳 🎉 Ты молодец!


Желаю тебе успехов в дальнейшем обучении. 😉

</div>